tests/chainer_tests/training_tests/extensions_tests/test_evaluator.py

```diff
@@ -215,4 +217,36 @@ def test_evaluate(self):
                 self.target.args[i], self.batches[i])
 
 
+@testing.parameterize(*testing.product({
+    'debug': [True, False],
+    'repeat': [True, False],
+    'iterator': [iterators.SerialIterator,
+                 iterators.MultiprocessIterator,
+                 iterators.MultithreadIterator]
+}))
+class TestEvaluatorRepeat(unittest.TestCase):
+
+    def setUp(self):
+        self.context = warnings.catch_warnings(record=True)
+        self.warnings = self.context.__enter__()
+        warnings.filterwarnings(action='always', category=DeprecationWarning)
+
+    def tearDown(self):
+        self.context.__exit__()
+
+    def test_serial_iterator(self):
+        dataset = numpy.ones((4, 6))
+        iterator = iterators.SerialIterator(dataset, 2, repeat=self.repeat)
+        with chainer.using_config('debug', self.debug):
+            with warnings.catch_warnings(record=True) as w:
+                extensions.Evaluator(iterator, {})
+
+        if self.debug and self.repeat:
+            expect = 1
+        else:
+            expect = 0
+
+        self.assertEqual(len(w), expect)
```

Comment:
Use `assert` (because we have moved to PyTest)

Reply:
Fixed